front/app/components/IdeaCards/IdeasWithFiltersSidebar/ButtonWithFiltersModal/index.tsx

```diff
@@ -0,0 +1,53 @@
+import React, { lazy, useState, Suspense } from 'react';
+
+import { Button, useBreakpoint } from '@citizenlab/cl2-component-library';
+
+import { FormattedMessage } from 'utils/cl-intl';
+
+import { Props as InputFiltersProps } from '../InputFilters';
+
+const FiltersModal = lazy(() => import('./FiltersModal'));
+import messages from './FiltersModal/messages';
+
+const ButtonWithFiltersModal = ({
+  selectedIdeaFilters,
+  onClearFilters,
+  ...filtersProps
+}: InputFiltersProps) => {
+  const isSmallerThanTablet = useBreakpoint('tablet');
+  const [filtersModalOpened, setFiltersModalOpened] = useState(false);
+
+  const openModal = () => {
+    setFiltersModalOpened(true);
+  };
+
+  const closeModal = () => {
+    setFiltersModalOpened(false);
+  };
+
+  if (!isSmallerThanTablet) return null;
+
+  return (
+    <>
+      <Button
```

Comment:
This is a best practice we should apply from now on. By moving modal's "opened state" in a separate component with the button that launches, we prevent all the surrounding components from re-rendering (which happened with the previous solution where e.g. the list of ideas would re-render each time we opened/closed the filters modal (even if we didn't change any of the filters).